pkg/operator/ceph/object/cosi/controller.go

```diff
@@ -0,0 +1,200 @@
+	if err != nil {
+		return fmt.Errorf("failed to create %s controller: %v", controllerName, err)
+	}
+
+	logger.Info("successfully started")
+	// Watch for changes to CephCOSIDriver
+	err = controller.Watch(source.Kind(mgr.GetCache(), &cephv1.CephCOSIDriver{}), &handler.EnqueueRequestForObject{}, opcontroller.WatchControllerPredicate())
+	if err != nil {
+		return fmt.Errorf("failed to watch for CephCOSIDriver object changes: %v", err)
+	}
+
+	// TODO: Watch for changes to CephObjectStore as arbitrary resource and predicate functions
+	err = controller.Watch(source.Kind(mgr.GetCache(), &cephv1.CephObjectStore{}), &handler.EnqueueRequestForObject{}, opcontroller.WatchControllerPredicate())
+	if err != nil {
+		return fmt.Errorf("failed to watch for CephObjectStore object changes: %v", err)
+	}
+
+	return nil
+}
+
+// Reconcile reads that state of the cluster for a CephCOSIDriver object and makes changes based on the state read
+// and what is in the CephCOSIDriver.Spec
+func (r *ReconcileCephCOSIDriver) Reconcile(context context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reconcileResponse, cephCOSIDriver, err := r.reconcile(request)
+
+	return reporting.ReportReconcileResult(logger, r.recorder, request, &cephCOSIDriver, reconcileResponse, err)
+}
+
+func (r *ReconcileCephCOSIDriver) reconcile(request reconcile.Request) (reconcile.Result, cephv1.CephCOSIDriver, error) {
+	cephCOSIDriver := &cephv1.CephCOSIDriver{}
+
+	// Fetch the CephCOSIDriver instance
+	err := r.client.Get(r.opManagerContext, request.NamespacedName, cephCOSIDriver)
+	logger.Debugf("CephCOSIDriver: %+v", cephCOSIDriver)
+	if err != nil && client.IgnoreNotFound(err) != nil {
+		return reconcile.Result{}, *cephCOSIDriver, fmt.Errorf("failed to get CephCOSIDriver. %+v", err)
+	}
+
+	cosiDeploymentStrategy := "auto"
+	if !reflect.DeepEqual(cephCOSIDriver.Spec, cephv1.CephCOSIDriverSpec{}) && cephCOSIDriver.Spec.DeploymentStrategy != "" {
+		cosiDeploymentStrategy = cephCOSIDriver.Spec.DeploymentStrategy
+	}
+	if cosiDeploymentStrategy == "disabled" {
+		logger.Info("Ceph COSI Driver is disabled")
+		return reconcile.Result{}, *cephCOSIDriver, nil
+	}
+
+	r.recorder.Eventf(cephCOSIDriver, kapiv1.EventTypeNormal, "Reconciling", "Reconciling Ceph COSI Driver %q", cephCOSIDriver.Name)
+	// Set the default CephCOSIDriver name if not already set
+	if cephCOSIDriver.Name == "" {
+		cephCOSIDriver.Name = CephCOSIDriverName
+	}
+
+	// Set the default CephCOSIDriver namespace if not already set
+	if cephCOSIDriver.Namespace == "" {
+		cephCOSIDriver.Namespace = os.Getenv(k8sutil.PodNamespaceEnvVar)
+	}
+
+	// Check whether object store is running
+	list := &cephv1.CephObjectStoreList{}
+	err = r.client.List(r.opManagerContext, list)
+	if err != nil {
+		return reconcile.Result{}, *cephCOSIDriver, fmt.Errorf("failed to list object stores. %+v", err)
+	}
+
+	// TODO: Check for object store part of Multisite
```

Comment:
Most like it should work IMO but not sure whether it any complications

Reply:
Ok then we can remove the TODO if we don't know of any needed work. Since it's an experimental feature, it's something we can fix in the future if any issue is found with multisite.